code/modules/mob/living/carbon/emote.dm

```diff
@@ -36,6 +42,28 @@
 				            'sound/misc/clap4.ogg')
 			playsound(user, clap, 50, 1, -1)
 
+/datum/emote/living/carbon/snap
+	key = "snap"
+	message = "snaps their fingers."
+	muzzle_ignore = TRUE
+	restraint_check = TRUE
+	emote_type = EMOTE_AUDIBLE
+
+/datum/emote/living/carbon/snap/run_emote(mob/living/user, params, playerInvoked = FALSE)
+	if (playerInvoked)
+		if (!isnull(cooldown_list[user.client.ckey]) && world.time < cooldown_list[user.client.ckey])
+			to_chat(user, "<span class='notice'>You cannot sound emote so soon.</span>")
+			return FALSE
+		else
+			cooldown_list[user.client.ckey] = world.time + cooldown
+	. = ..()
+	if (.)
+		if (ishuman(user))

```

Comment:
make a subemote at the mob/living/carbon/human and check this there instead, or add a can_snap proc